src/PostgREST/App.hs

```diff
@@ -419,7 +420,8 @@ handleInfo identifier RequestContext{..} =
     Just table ->
       return $ Wai.responseLBS HTTP.status200 [allOrigins, allowH table] mempty
     Nothing ->
-      throwError Error.NotFound
+      -- TODO is this right? When no tbl is found on the schema cache we disallow OPTIONS?
+      throwError $ Error.ApiRequestError ApiRequestTypes.NotFound
```

Comment:
I think that's the intended response: 404 on a non existent resource. Also, this line was added long ago, according to the [blame history](https://github.com/PostgREST/postgrest/blame/1eda7db0fbbdb4bde5621414ede241b8f30e213e/src/PostgREST/App.hs#L285).

Reply:
Hm, wait. We talked about this on https://github.com/PostgREST/postgrest/pull/1839#issuecomment-829386914. Of course GET/POST should be allowed but now that we restrict PATCH/DELETE, we cannot include them in the allow unless they include a filter.

This would be better left for another PR.